src/Particular.Analyzers/DiagnosticDescriptors.cs

```diff
@@ -171,5 +171,21 @@ public static class DiagnosticDescriptors
             category: "Code",
             defaultSeverity: DiagnosticSeverity.Warning,
             isEnabledByDefault: true);
+
+        public static readonly DiagnosticDescriptor DateTimeAssignedToDateTimeOffset = new DiagnosticDescriptor(
+            id: DiagnosticIds.DateTimeAssignedToDateTimeOffset,
+            title: "Prevent a DateTime from being implicitly assigned to a DateTimeOffset",
+            messageFormat: "Do not implicitly assign a DateTime value to a DateTimeOffset variable",
```

Comment:
Related to my other comment about the name of the analyzer:

```suggestion
            messageFormat: "Do not implicitly cast a DateTime to a DateTimeOffset",
```